aws-bundle/build.gradle

```diff
@@ -36,6 +38,9 @@ project(":iceberg-aws-bundle") {
     implementation "software.amazon.awssdk:sts"
     implementation "software.amazon.awssdk:dynamodb"
     implementation "software.amazon.awssdk:lakeformation"
+    implementation "software.amazon.awssdk:url-connection-client"
```

Comment:
I think the sync client defaults to apache client, so we don't need to add this URL connection client

Reply:
Sure, I'm happy to remove it - I have added because it's present [here](https://github.com/apache/iceberg/blob/main/build.gradle#L462):  Should we consider removing there too?

Reply:
Thats a compile only addition in `iceberg-aws` to specifically not include it in the runtime. Here we are including in the runtime. So, I am inclined towards removing this. 

@jackye1995, @nastra : WDYT?

Reply:
Yeah, I agree. Removed it now..